### DISCUSSION TITLE
怎么配合1panel的openresty食用呢

### DISCUSSION THREAD

**Discussion by Felmon** [chaitin/SafeLine#417] [Q&A]
### 反馈内容

目前在用1panel面板，搭建了不少网站，但是自带的waf有些功能不完美（开发组那边也建议waf部分功能使用第三方waf实现），想用长亭waf，但是两个应用都占用80 443端口且都涉及nginx，怎么搭配openresty建站和长亭的waf功能呢？比如不占用端口可以批量导入1panel opnresty的站点配置来安全守护1panel搭建的网站呢

**Comment by luweijun1992** [comment]
分成2台虚拟机构建，就不存在端口占用。
如果非要同一台服务器上，就用不同端口把。

**Comment by Lorna0** [comment]
官网文档中有比较详细的说明了：https://waf-ce.chaitin.cn/docs/guide/config 。简而言之分两台部署最好，实在要在同一台设备上部署就要改端口，咋改都行，只要流量过雷池就可以防护。

**Comment by zclaiqcc** [comment]
> ### 反馈内容
> 目前在用1panel面板，搭建了不少网站，但是自带的waf有些功能不完美（开发组那边也建议waf部分功能使用第三方waf实现），想用长亭waf，但是两个应用都占用80 443端口且都涉及nginx，怎么搭配openresty建站和长亭的waf功能呢？比如不占用端口可以批量导入1panel opnresty的站点配置来安全守护1panel搭建的网站呢

看到 @Felmon 师傅在 1Panel 那边的 issue 和评论了。简单说 80/443 端口占用问题确实是面板和 WAF 配合使用的一个比较头疼的问题。

具体的接入方式我粗略整理了一下：
1. 两台机器部署，一了百了
2. 同一台机器部署，OpenResty 监听 80/443，回源地址 upstream 填雷池 WAF 上监听的端口，在雷池上配置「上游服务器」到师傅的 Web 服务。
    - 相当于是「OpenResty 80/443」--->「WAF 8080/8443」--->「业务服务器 8888/7443」（端口我随便写的）
3. 同一台机器部署，通过 [lua-t1k](https://github.com/chaitin/lua-resty-t1k) 的方式，但是需要师傅自己 DIY 一下。比如可以参考 https://zhuanlan.zhihu.com/p/641117894
    - 相当于是「OpenResty 80/443」--->「WAF 不占用额外端口」--->「业务服务器 8888/7443」（端口我随便写的）
4. 同一台机器部署，修改 OpenResty 出让 80/443 端口，这样似乎会也失去了面板管理的功能，因为 80/443 被雷池占用了
    - 相当于是「WAF 80/443」--->「OpenResty 8080/8443」--->「业务服务器 8888/7443」（端口我随便写的）

**Comment by Felmon** [comment]
> > ### 反馈内容
> > 目前在用1panel面板，搭建了不少网站，但是自带的waf有些功能不完美（开发组那边也建议waf部分功能使用第三方waf实现），想用长亭waf，但是两个应用都占用80 443端口且都涉及nginx，怎么搭配openresty建站和长亭的waf功能呢？比如不占用端口可以批量导入1panel opnresty的站点配置来安全守护1panel搭建的网站呢
> 
> 看到 @Felmon 师傅在 1Panel 那边的 issue 和评论了。简单说 80/443 端口占用问题确实是面板和 WAF 配合使用的一个比较头疼的问题。
> 
> 具体的接入方式我粗略整理了一下：
> 
> 1. 两台机器部署，一了百了
> 2. 同一台机器部署，OpenResty 监听 80/443，回源地址 upstream 填雷池 WAF 上监听的端口，在雷池上配置「上游服务器」到师傅的 Web 服务。
>    
>    * 相当于是「OpenResty 80/443」--->「WAF 8080/8443」--->「业务服务器 8888/7443」（端口我随便写的）
> 3. 同一台机器部署，通过 [lua-t1k](https://github.com/chaitin/lua-resty-t1k) 的方式，但是需要师傅自己 DIY 一下。比如可以参考 https://zhuanlan.zhihu.com/p/641117894
>    
>    * 相当于是「OpenResty 80/443」--->「WAF 不占用额外端口」--->「业务服务器 8888/7443」（端口我随便写的）
> 4. 同一台机器部署，修改 OpenResty 出让 80/443 端口，这样似乎会也失去了面板管理的功能，因为 80/443 被雷池占用了
>    
>    * 相当于是「WAF 80/443」--->「OpenResty 8080/8443」--->「业务服务器 8888/7443」（端口我随便写的）

那1panel那边搭建的php网站要怎么用呢？有个php-fpm端口，上游填这个端口没用

**Reply by skyhhjmk** [reply]
不理解你在搞什么，php-fpm端口是openresty和php容器通信的端口，填这个干嘛

**Comment by zclaiqcc** [comment]
> 那1panel那边搭建的php网站要怎么用呢？有个php-fpm端口，上游填这个端口没用

这个不确定了，师傅搜一下 NGINX 代理 php-fpm

**Comment by Anyexyz** [comment]
https://blog.csdn.net/qq_20003465/article/details/134768530
https://blog.csdn.net/qq_20003465/article/details/135501249
请佬参考
